studio/components/interfaces/Billing/ProUpgrade.tsx

```diff
@@ -133,6 +133,16 @@ const ProUpgrade: FC<Props> = ({
     getSubscriptionPreview()
   }, [selectedComputeSize, selectedPITRDuration, selectedCustomDomainOption, isSpendCapEnabled])
 
+  // this is used to navigate to addon hash anchors on first load, allows us to directly link subscription addons
+  useEffect(() => {
+    if (router.isReady) {
+      const hash = router.asPath.split('#')[1]
+      if (hash !== undefined) {
+        router.replace({ hash }, null, { shallow: true })
```

Comment:
Replaced the old logic below as it messes up the router history where users can't navigate back anymore

```
    if (hash !== undefined) {
      window.location.hash = ''
      window.location.hash = hash
    }
```